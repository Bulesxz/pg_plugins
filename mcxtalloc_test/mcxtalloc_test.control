# mcxtalloc_test extension
comment = 'Minimal Extension Template'
default_version = '1.0'
module_pathname = '$libdir/mcxtalloc_test'
relocatable = true
