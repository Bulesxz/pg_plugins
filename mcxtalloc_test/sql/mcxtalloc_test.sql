--
-- Test for mcxtalloc_test
--

-- Depending on the environment where this test suite is run, the output may
-- defer.

-- Initialize environment
CREATE EXTENSION mcxtalloc_test;

-- Error, incorrect size
SELECT mcxtalloc(1024 * 1024 * 1024);
-- May fail because of OOM
SELECT mcxtalloc(1024 * 1024 * 1024 - 1);

-- May fail because of OOM, size correct
SELECT mcxtalloc_huge(1024 * 1024 * 1024);

-- Should return true
SELECT mcxtalloc_zero_cmp(1024);

-- Always failure because of incorrect size, memory requested is higher than 1GB
SELECT mcxtalloc_extended(1024 * 1024 * 1024, false, false, false);
-- Success because size is fine, or failure with OOM
SELECT mcxtalloc_extended(1024 * 1024 * 1024 - 1, false, false, false);
-- Always success, size is correct, and OOM is bypassed
-- Returns false if OOM occurred as allocated pointer was NULL.
SELECT mcxtalloc_extended(1024 * 1024 * 1024 - 1, false, true, false);
-- Always success, huge size is correct, and OOM is bypassed
-- Returns false if OOM occurred as allocated pointer was NULL.
SELECT mcxtalloc_extended(1024 * 1024 * 1024, true, true, false);
