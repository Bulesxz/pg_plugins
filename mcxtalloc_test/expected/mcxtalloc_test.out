--
-- Test for mcxtalloc_test
--
-- Depending on the environment where this test suite is run, the output may
-- defer.
-- Initialize environment
CREATE EXTENSION mcxtalloc_test;
-- Error, incorrect size
SELECT mcxtalloc(1024 * 1024 * 1024);
ERROR:  invalid memory alloc request size 1073741824
-- May fail because of OOM
SELECT mcxtalloc(1024 * 1024 * 1024 - 1);
 mcxtalloc 
-----------
 
(1 row)

-- May fail because of OOM, size correct
SELECT mcxtalloc_huge(1024 * 1024 * 1024);
 mcxtalloc_huge 
----------------
 
(1 row)

-- Should return true
SELECT mcxtalloc_zero_cmp(1024);
 mcxtalloc_zero_cmp 
--------------------
 t
(1 row)

-- Always failure because of incorrect size, memory requested is higher than 1GB
SELECT mcxtalloc_extended(1024 * 1024 * 1024, false, false, false);
ERROR:  invalid memory alloc request size 1073741824
-- Success because size is fine, or failure with OOM
SELECT mcxtalloc_extended(1024 * 1024 * 1024 - 1, false, false, false);
 mcxtalloc_extended 
--------------------
 t
(1 row)

-- Always success, size is correct, and OOM is bypassed
-- Returns false if OOM occurred as allocated pointer was NULL.
SELECT mcxtalloc_extended(1024 * 1024 * 1024 - 1, false, true, false);
 mcxtalloc_extended 
--------------------
 t
(1 row)

-- Always success, huge size is correct, and OOM is bypassed
-- Returns false if OOM occurred as allocated pointer was NULL.
SELECT mcxtalloc_extended(1024 * 1024 * 1024, true, true, false);
 mcxtalloc_extended 
--------------------
 t
(1 row)

